lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -755,10 +755,22 @@ def add_metadata(post, options = {})
         post[:metadata].merge!(options[:metadata]) if options[:metadata]
       end
 
-      def parse(body)
-        return {} if body.blank?
+      def parse(response)
+        return {} if response.blank?
```

Comment:
Couple thoughts here. I think that the parse call should still take body as the argument since that is what is being passed in, and we are adding the headers. Then I think that we should add the headers under their own tag in the response so that they aren't just mixed into the parsed body without any differentiation. 

I don't see any functional problems with the implementation though.

```suggestion
      def parse(body)
        return {} if body.blank?
        
        response = JSON.parse(body)
        response.merge!({response_headers: @response_headers}) if @response_headers.present?
        response
```

Reply:
thanks @DustinHaefele for the reccomendation! I was wondering would it be better if we specifically return only the field that is being requested "transient-error" or return the and merge all header fields. Would there be any security related risk to returning the all of the header fields? i was mulling over this when working through this however i'd rather return the all the fields in case we need to grab them in the future.

Reply:
Yeah that is a good point to. I don't think there is much risk, but since it would be pretty easy to add more headers in the future if others are requested maybe it is a good idea to only grab the one that we actually care about.